community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/storage/PartitionedIndexStorage.java

```diff
@@ -235,7 +246,10 @@ private ZipOutputStream archiveFile( File folder, boolean archiveFailed ) throws
     {
         File[] files = fileSystem.listFiles( rootFolder );
         return files == null ? Collections.emptyList()
-                             : Stream.of( files ).filter( fileSystem::isDirectory ).collect( toList() );
+                             : Stream.of( files )
+                               .filter( fileSystem::isDirectory )
+                               .sorted( FILE_COMPARATOR )
```

Comment:
how would that work? NumberAwareStringComparator compares Strings, this thing compares Files